### ISSUE TITLE
warning if compiling and linking using [-no]-pie 

### ISSUE THREAD

**Issue by nolange** [llvm/llvm-project#95108]
When linking for windows you will get false warnings, see:

``` bash
echo "int main() {return 0;}" | clang++ --target=x86_64-windows-gnu -x c++ -pie -
```

Will result in:

```
clang++: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
```

Despite the clang invocation doing linking. When targeting Linux there is no warning.

Using Clang 18.1.6 on Debian x86_64.
